test/util.js

```diff
@@ -513,3 +515,261 @@ function checkCursorPaginatedHistoryObject ({
+ * @param {Boolean}  [params.filters[i].isArrayFilter = false]
+ * @param {Boolean}  [params.filters[i].isRangeFilter = false]
+ * @param {Object[]} [params.filters[i].customTestValues] - if specified, will only use those values
+ *   and test each one of those values
+ *   otherwise, will perform a fetch of objects and test with fixtures values
+ * @param {Boolean}  [params.filters[i].noResultsExistenceCheck = false]
+ *   by default, will check if there are results while it should have (based on auto-fetched values)
+ */
+async function checkFilters ({
+  t,
+  authorizationHeaders,
+  endpointUrl,
+  fetchEndpointUrl,
+  checkPaginationObject,
+  filters,
+}) {
+  const getEndpoint = (endpointUrl, queryParams) => {
+    if (endpointUrl.includes('?')) return `${endpointUrl}&${queryParams}`
+    else return `${endpointUrl}?${queryParams}`
+  }
+
+  const getRandomValue = (values) => _.first(_.shuffle(values))
+
+  const sortValues = (values) => {
+    return values.slice(0).sort((a, b) => {
+      if (a < b) return -1
+      else if (a > b) return 1
+      else return 0
+    })
+  }
+
+  // prop=value
+  const getExactValueFilter = (prop, exactValue) => `${prop}=${exactValue}`
+
+  // prop[]=value1&prop[]=value2
+  const getArrayValuesFilter = (prop, values) => values.map(v => `${prop}[]=${v}`).join('&')
+
+  // prop=value1,value2
+  const getArrayCommaSeparatedValuesFilter = (prop, values) => `${prop}=${values.join(',')}`
+
+  // prop[gte]=min&prop[lte]=max
+  const getRangeFilter = (prop, range) => `${prop}[gte]=${range.gte}&${prop}[lte]=${range.lte}`
+
+  const addCheckObjFunctions = (filter) => {
+    const {
+      prop,
+      customGetValue,
+      isArrayFilter,
+      isRangeFilter,
+      customCheck
+    } = filter
+
+    const getValue = (obj) => _.isFunction(customGetValue) ? customGetValue(obj) : obj[prop]
+    filter.getValue = getValue
+
+    filter.getExactValueCheckObj = _.curry((exactValue, obj) => {
+      return _.isFunction(customCheck)
+        ? customCheck(obj, exactValue)
+        : getValue(obj) === exactValue
+    })
+
+    if (isArrayFilter) {
+      filter.getArrayValuesCheckObj = _.curry((values, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, values)
+          : values.includes(getValue(obj))
+      })
+    } else if (isRangeFilter) {
+      filter.getRangeValuesCheckObj = _.curry((rangeValues, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, rangeValues)
+          : rangeValues.gte <= getValue(obj) && getValue(obj) <= rangeValues.lte
+      })
+    }
+  }
+
+  const addTestValues = (filter) => {
+    const {
+      isArrayFilter,
+      isRangeFilter,
+      customTestValues,
+    } = filter
+
+    addCheckObjFunctions(filter)
+
+    const getValue = filter.getValue
+
+    const useCustomValues = customTestValues && customTestValues.length
+
+    const realValues = _.compact(_.uniq(results.map(getValue)))
```

Comment:
The scope for `results` seems to be implicit (strong code smell).

This whole `checkFilters` function is quite contrieved and the vague JS doc comment doesn’t help.

Reply:
Thanks for the catch. `results` must be passed as parameter to the function `addTestValues`. This change was needed after some refactor when implementing the util function.

I'll add more details on how to use this util and how it should be have.